File: ruby/src/Creational/Builder/HonoluluSoupBuffetBuilder.rb


#/HonoluluSoupBuffetBuilder - Two of Two Builder Subclasses

require "Creational/Builder/SoupBuffetBuilder"
require "common/HonoluluClamChowder"
require "common/HonoluluFishChowder"

class HonoluluSoupBuffetBuilder < SoupBuffetBuilder

   def initialize
   end

	def buildClamChowder()
   		@soupBuffet.clamChowder = HonoluluClamChowder.new
	end

	def buildFishChowder()
   		@soupBuffet.fishChowder = HonoluluFishChowder.new
	end

	def setSoupBuffetName()
 		@soupBuffet.soupBuffetName = "Honolulu Soup Buffet"
	end

end

File: csharp/src/Creational/Builder/HonoluluSoupBuffetBuilder.cs
//HonoluluSoupBuffetBuilder.java - Two of Two Builder Subclasses

namespace Creational.Builder {

using common;

class HonoluluSoupBuffetBuilder : SoupBuffetBuilder {
    public new void buildClamChowder() {
        soupBuffet.clamChowder = new HonoluluClamChowder();
    }
    public new void buildFishChowder() {
        soupBuffet.fishChowder = new HonoluluFishChowder();
    }
    
    override public void setSoupBuffetName() {
        soupBuffet.soupBuffetName = "Honolulu Soup Buffet";
    }
}

}
